src/api/web3Provider.ts

```diff
@@ -0,0 +1,63 @@
+import { ProviderInterface } from './types'
+import { windowLoaded, promisify } from './utils'
+import { Web3 } from 'web3'
+
+const getProvider = () => {
+  if (typeof window !== 'undefined' && window.web3) {
+    return window.web3.currentProvider
+  } else {
+    return new Web3.providers.HttpProvider('http://localhost:8545')
+  }
+}
+
+const setupWeb3 = async () => {
+  await windowLoaded
+
+  return new Web3(getProvider())
+}
+
+export const promisedWeb3 = init()
+
+async function init(): Promise<ProviderInterface> {
+  const web3 = await setupWeb3()
+
+  const getAccounts = promisify(web3.eth.getAccounts, web3.eth)
+  const getBalance = promisify(web3.eth.getBalance, web3.eth)
+
+  const getCurrentAccount = async () => {
+    const [account] = await getAccounts()
+
+    return account
+  }
+
+  const getETHBalance = async () => {
+
+    const account = await getCurrentAccount()
+
+    const wei = await getBalance(account)
+
+    return web3.fromWei(wei, 'ether')
+  }
+
+  const getNetwork = promisify(web3.eth.getNetwork, web3.eth)
+
+  const isConnected = web3.isConnected.bind(web3)
+
+  const setProvider = web3.setProvider.bind(web3)
+
+  const resetProvider = () => setProvider(getProvider())
+
+  return {
+    getCurrentAccount,
+    getAccounts,
+    getETHBalance,
+    getNetwork,
+    isConnected,
+    get currentProvider() {
```

Comment:
This is new to me - are you declaring the function here _and_ calling the getter?

Reply:
declaring the getter, same as
```
class CL {
  get prop () {
    return this.prop
  }
}
```